bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/dialogs/FilteredTree.java

```diff
@@ -788,6 +793,16 @@ protected void setFilterText(String filterText) {
 		}
 	}
 
+	/**
+	 * Override the default refresh job delay
+	 *
+	 * @param time
+	 * @since 3.132
+	 */
+	public void setRefreshJobDelay(long time) {
+		this.refreshJobDelay = time;
+	}
```

Comment:
I would propose to provide this as an instantiation parameter (maybe as a contructor parameter or, even better, via some factory method) and make the stored value immutable. The issue with making this value mutable is that you have to consider the cases where this value has been accessed and has a current effect, in particular in terms of a pending delay. So you need to be precise about the effects of setting this value after instantiation. Making the value immutable solves all these problems.

As an additional note: the public API should not expose an information about a "refresh job". The public API could state that there is some delay after which the tree contents are refreshed, but doing that via a job is an implementation detail that should be hidden from the public API.

Reply:
Has this been addressed? I think you meant that the name of the setter should not have the word `Job` in it, right? But the setter is gone, the new parameters do not have the word `Job` in it and the new field (and its JavaDoc) are `private`, so it should be fine, right?

Reply:
@fedejeanne yes, I have removed the setter and used constructor to initialize the final value. Also updated the JavaDoc without the word job in it.

Reply:
Yes, my original comment has been addressed, but unfortunately I don't think this is ready to be merged yet (see my recent comments).